sphinx/environment.py

```diff
@@ -961,6 +962,29 @@ def process_downloads(self, docname, doctree):
 
     def process_images(self, docname, doctree):
         """Process and rewrite image URIs."""
+        def collect_candidates(imgpath, candidates):
+            globbed = {}
+            for filename in glob(imgpath):
+                new_imgpath = relative_path(path.join(self.srcdir, 'dummy'),
+                                            filename)
+                if filename.lower().endswith('.pdf'):
+                    if 'application/pdf' not in candidates:
+                        globbed.setdefault('application/pdf', []).append(new_imgpath)
```

Comment:
Oh, sorry. I misread your comments. I agree with you.
But, it is already refactored by other comment like following:
```
        def collect_candidates(imgpath, candidates):
            globbed = {}
            for filename in glob(imgpath):
                rel_imgpath = relative_path(path.join(self.srcdir, 'dummy'),
                                            filename)
                mimetype = guess_mimetype(filename)
                if mimetype and mimetype not in candidates:
                    globbed.setdefault(mimetype, []).append(rel_imgpath)
            for key, files in iteritems(globbed):
                candidates[key] = sorted(files, key=len)[0]  # select by similarity
```